aten/src/ATen/detail/FunctionTraits.h

```diff
@@ -76,3 +76,12 @@ struct binary_function_traits {
   using arg1_t = typename traits::template arg<0>::type;
   using arg2_t = typename traits::template arg<1>::type;
 };
+
+template <typename T>
+struct ternary_function_traits {
```

Comment:
Please don't add a `ternary_function_traits`, it is being removed in https://github.com/pytorch/pytorch/pull/35774, see https://github.com/pytorch/pytorch/pull/35774#discussion_r409935812